shared/src/main/scala/pl/metastack/metatime/DateTime.scala

```diff
@@ -1,18 +1,87 @@
 package pl.metastack.metatime
 
-trait DateTime {
-  def date: Date = ???
-  def format: String = ???
+trait DateTime extends Time with Date {
   def format(fmt: String): String = ???
   def timestamp: Timestamp = ???
-  def -(time: Component): DateTime = ???
-  def +(time: Component): DateTime = ???
-  def -(dateTime: DateTime): DateTime = ???
-  def +(dateTime: DateTime): DateTime = ???
-  def fromNow: Offset[DateTime] = ???
+
+  override def equals(that: Any): Boolean =
+    that match {
+      case other: DateTime => other.year == year && other.month == month && other.day == day &&
+        h == other.h && m == other.m && s == other.s && ms == other.ms
+      case _ => false
+    }
+
+  override def unix(): Unix = Unix(
+    Year(year).unix().value +
+    Month(month).unix().value +
+    Day(day).unix().value +
+    Hour(h).unix().value +
+    Minute(m).unix().value +
+    Second(s).unix().value +
+    Millisecond(ms).unix().value)
 }
 
 object DateTime {
-  def now(): DateTime = ???
-  def apply(year: Int, month: Int, day: Int): DateTime = ???
+
+  def calculateDateTime(milliDate: Long, milliTime: Long): DateTime = {
+    val date = Date(milliDate)
+    val time = Time(milliTime)
+    DateTime(date.year, date.month, date.day, time.h, time.m, time.s, time.ms).asInstanceOf[DateTime]
```

Comment:
Change the `apply()` method to return the right type (`DateTime` instead of `Date`).